### PR TITLE
Document genreconciler.

### PR DESCRIPTION
**Pull request by n3wscott**
Adding documentation on how to use and what to expect from the generated reconciler code added to client injection gen.

/assign @mattmoor

### CODE REVIEW

File: injection/README.md [hunk 1]
```diff
@@ -216,3 +280,108 @@ required = [
...
+    `Reconciler`.
+- If additional events are required to be produced, an implementation can pull a
+  recorder from the context: `recorder := controller.GetEventRecorder(ctx)`.
+
+Pending Features:
+
+- Leverage `configStore` and specifically `ctx = r.configStore.ToContext(ctx)`
+  inside `Reconcile`.
+- Resulting changes from `Reconcile` calling `ReconcileKind(ctx, resource)`:
+  - If resource.status.finalizers are updated, `Reconcile` will synchronize it
```
Comment (mattmoor) [inline injection/README.md]: This is done.

File: injection/README.md [hunk 2]
```diff
@@ -216,3 +280,108 @@ required = [
...
+    back to the API Server.
+  - If `resource.metadata.labels` or `.annotations` are updated, `Reconcile`
+    will synchronize it back to the API Server.
```
Comment (mattmoor) [inline injection/README.md]: Do we really want to?

File: injection/README.md [hunk 3]
```diff
@@ -33,6 +33,78 @@ func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl
...
+If generated reconcilers are used, Type `Reconciler` is expected to implement
+`ReconcileKind`:
+
+```go
+func (r *Reconciler) ReconcileKind(ctx context.Context, o *samplesv1alpha1.AddressableService) reconciler.Event {
+    ...
+}
+```
+
+And if finalizers are required,
```
Comment (mattmoor) [inline injection/README.md]: Remove trailing whitespace:
```suggestion
And if finalizers are required,
```

File: injection/README.md [hunk 4]
```diff
@@ -216,3 +288,133 @@ required = [
...
+  - Additionally, the `error` will be returned from `Reconciler` and `key` will
+    requeue back into the reconciler key queue.
+- If `event` is a `reconciler.Event`, `Reconciler` will log a typed and reasoned
+  Kubernetes Event based on the contents of `event`.
+  - `event` is not considered an error for requeue and nil is returned from
+    `Reconciler`.
+- If additional events are required to be produced, an implementation can pull a
+  recorder from the context: `recorder := controller.GetEventRecorder(ctx)`.
+
+Future features to be concidered:
```
Comment (mattmoor) [inline injection/README.md]: Found potential misspelling(s):
```suggestion
Future features to be considered:
```

**Comment by n3wscott** [comment]
@googlebot I consent.

**Comment by n3wscott** [comment]
@googlebot I fixed it.

**Pull request merged**